### ISSUE TITLE
Document how to use the getHeader method

### ISSUE THREAD

**Issue by kevinburkeshyp** [request/request#1760]
I'm trying to use it to retrieve a header from the response, but I haven't been able to figure out how to do it.

**Comment by shaochun** [comment]
request.get({uri: url, }, function (error, response, body) {
  if (!error && response.statusCode == 200) {
    console.log(response.headers);
  }
}

**Comment by kevinburke** [comment]
Right... I guess my worry there is that since `response.headers` is a JS Object, I can search for `response.headers['Cache-Control']` but I would miss a header sent back that was named `'cache-control'`. Per the spec, HTTP headers are supposed to case insensitive.

**Comment by simov** [comment]
@kevinburke you can use [caseless](https://www.npmjs.com/package/caseless) in your project.

**Comment by shaochun** [comment]
@simov great to know, thanks.

**Comment by csaroff** [comment]
> I guess my worry there is that since response.headers is a JS Object, I can search for response.headers['Cache-Control'] but I would miss a header sent back that was named 'cache-control'. Per the spec, HTTP headers are supposed to case insensitive.

Not sure if this is a new feature, but request lowercases all headers before placing them in the headers object.  If you receive the header `'Cache-Control'`,  it will be located under `response.headers['cache-control']`.  No need to worry about an extra module like caseless :)

I think this issue can be closed.

**Comment by sahildua2305** [comment]
@csaroff If you are using `request` module, `caseless` is anyway being used as a dependency. So, in a way you are already using it. So it won't make a difference if you use `caseless` to get headers as well.

Just like this:

```javascript
console.log(response.caseless.get('cache-control'))
```
